backend/Api/Staffing/ConsultantController.cs

```diff
@@ -39,69 +40,51 @@ public ConsultantController(ApplicationContext context, IMemoryCache cache)
-                service.UpdateStaffing(staffingId, hours);
-                break;
-            case BookingType.PlannedAbsence:
-                service.UpdateAbsence(staffingId, hours);
-                break;
-            case BookingType.Vacation:
-                break;
-            default:
-                throw new ArgumentOutOfRangeException(nameof(bookingType), bookingType, "Invalid bookingType");
+            return BadRequest();
         }
 
-        return Ok(hours);
-    }
-    
-    [HttpPost]
-    [Route("staffing/new")]
-    public ActionResult<List<ConsultantReadModel>>Post(
-        [FromRoute] string orgUrlKey,
-        [FromQuery(Name = "Type")] BookingType bookingType,
-        [FromQuery(Name = "ConsultantID")] int consultantId,
-        [FromQuery(Name = "EngagementID")] int engagementId,
-        [FromQuery(Name = "Year")] int selectedYearParam,
-        [FromQuery(Name = "Week")] int selectedWeekParam,
-        [FromQuery(Name = "Hours")] double hours = 0
-    )
-    {
-        var service = new StorageService(_cache, _context);
+        try
+        {
+            var selectedWeek = new Week(staffingWriteModel.Year, staffingWriteModel.Week);
 
-        var newId = 0;
-        
-        var selectedWeek = new Week((int)selectedYearParam, (int)selectedWeekParam);
+            switch (staffingWriteModel.Type)
+            {
+                case BookingType.Booking:
+                case BookingType.Offer:
+                    service.UpdateOrCreateStaffing(new StaffingKey(staffingWriteModel.EngagementId, staffingWriteModel.ConsultantId, selectedWeek), staffingWriteModel.Hours, orgUrlKey);
+                    break;
+                case BookingType.PlannedAbsence:
+                    service.UpdateOrCreatePlannedAbsence(new PlannedAbsenceKey(staffingWriteModel.EngagementId, staffingWriteModel.ConsultantId, selectedWeek), staffingWriteModel.Hours, orgUrlKey);
+                    break;
+                case BookingType.Vacation:
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(staffingWriteModel.Type), staffingWriteModel.Type, "Invalid bookingType");
+            }
 
-        switch (bookingType)
+        }
+        catch (Exception e)
         {
-            case BookingType.Booking:
-            case BookingType.Offer:
-                newId = service.CreateStaffing(consultantId, engagementId, hours, selectedWeek );
-                break;
-            case BookingType.PlannedAbsence:
-                newId = service.CreateAbsence(consultantId, engagementId, hours, selectedWeek );
-                break;
-            case BookingType.Vacation:
-                break;
-            default:
-                throw new ArgumentOutOfRangeException(nameof(bookingType), bookingType, "Invalid bookingType");
+            Console.WriteLine(e);
+            throw;
         }
 
-        return Ok(newId);
+
+        return Ok(0);
```

Comment:
What kind of response is this? just:
```
{
"0" ? 
}

Reply:
A really lazy one. Frontend does not nessesarily need a response other than «Ok» but next expects a response it can use json parse on. And instead of rewriting updateandcreate or the api call on the frontend side I did this 😅 I can change it tomorrow or write a comment if you want

Reply:
I dont understand if tumbs up means "change it"/"comment" or it´s good as it is 😅

Reply:
I did manage to change the frontend code so it does not crash, so i can change the response to just be `Ok()`

Reply:
Skratch that, it still complains in the console.log

Reply:
If you just have `Ok()`, the body will be completely empty, or `null`
So, frontend this line will give an error: `response.json()`

We can solve it in two-three ways. 
Frontend: Check if the type T used in the generic `callApi` is `never, null or undefined` etc. Then skip `response.json`
Frontend: Just return the response, and make every individual call choose to call `response.json()`
Backend: Provide a blank, but non-null reponse like 0, [], {} or something

Reply:
We decided to return a string with "Status Ok" 😅

Reply:
Cool :)